api/proto/auth/client.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2015-2020 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package auth
+
+import (
+	"sync"
+	"sync/atomic"
+
+	"github.com/gravitational/roundtrip"
+	"github.com/gravitational/teleport"
+	"github.com/gravitational/trace"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+)
+
+var log = logrus.WithFields(logrus.Fields{
+	trace.Component: teleport.ComponentClient,
+})
+
+// Client is a gRPC Client that connects to a teleport auth server through TLS.
+type Client struct {
+	sync.Mutex
```

Comment:
I'm fine with removing the lazy loading, unless there is a reason to maintain the lazy loading functionality from the original implementation of the client in `clt.go`? @russjones

Reply:
I added GRPC lazy loading long time ago when we only had one method using GRPC and I wanted to avoid having an extra connection with no reason. Now when GRPC is a default we can remove the lazy load and init as awly suggests